geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -198,12 +198,14 @@ object FilterHelper {
   def extractAttributeBounds[T](filter: Filter, attribute: String, binding: Class[T]): FilterValues[Bounds[T]] = {
     filter match {
       case o: Or =>
+        val set = o.getChildren.foldLeft(Set.empty[Option[String]])((acc, f) => acc + getAttributeProperty(f))   
```

Comment:
i think the logic here is slightly more restrictive than we need it to be. for example, something like:
`(dtg DURING a/b AND geom bbox) OR (dtg DURING c/d)` would fail this check, however, we could still extract `dtg DURING a/b OR dtg DURING c/d` as a valid bounds.

I think instead, the condition that we want to give up on is if any of the children bounds are empty, e.g. in the case of `dtg DURING a/b OR geom bbox`, the geom clause would return an empty bounds. we could modify the `o.getChildren.flatMap` to be a `o.getChildren.reduce` of some sort (or maybe a .map followed by a .fold), and give up and return an empty filter values if we hit a child clause that returns an empty bounds.

It would be good to add those 2 cases as a unit test, too.

Reply:
Ah ok, makes sense. I'll make those changes